### DISCUSSION TITLE
모수팀이 생각하는 좋은 코드란 무엇일까요?

### DISCUSSION THREAD

**Discussion by Daae-Kim** [MOSOO-team/MOSOO-BE#5] [General]
모수 프로젝트 리팩토링에 앞서, 기존 코드 개선 및 일관성이 있는 프로젝트 코드를 위해 모수팀에서 정의하는 좋은 코드에 대해 논의하면 좋을것 같아 모수팀의 첫 디스커션을 열었습니다.

창민님이 생각하시는 좋은 코드란 무엇인가요?
우선, 제가 생각해오고 있는 좋은코드란
1. 한개의 메소드는 한개의 역할을 한다. 
2. 남이 보기에 읽기 쉬워야한다.  (if else 문 -> if 로 나타내기, 0과 1에 의미를 담지 않기. 남이 보기에 쉬워야 나중에 자신이 보았을때도 쉽기 때문)
3. 객체의 역할이 명확하다.
입니다. 이에 대한 창민님의 생각과 창민님만의 기준이 궁금합니다 ㅎㅎ @Engineering-Student-An

**Comment by anchangmin35** [comment]
저도 비슷하게 생각하고 있습니다!
다애님이 말씀하신 3가지 모두 동의하며 손색이 없다고 생각합니다. 

비슷한 내용인 것 같긴 하지만 굳이 기준을 추가해본다면 
1. 중복되는 코드를 없애고 재사용성을 높인다.
2. 변수명, 메서드 명등을 명확하게 짓고 적절한 주석을 추가한다.
3. 관심사를 분리하고 기능별로 코드를 분리해 모듈화한다.
정도가 될 것 같습니다!

**Reply by Daae-Kim** [reply]
그동안 창민님 코드를 봐온입장으로서, 창민님 코드에 명확히 나타나는 기준들인것 같아요~!! 
혹시 "변수명, 메소드명을 명확하게 짓는다" 에 대해서 조금 더 자세한 설명을 들을 수 있을것 같은데, 명확한 변수명, 메소드명을 짓는다는것은 
동사위주의 메소드명으로 적고, 역할의 의도가 들어나는 단어를 사용하는것을 말씀하시는게 맞을까요?
추가로 적절한 주석이라는것이 어떤것을 의미하는지도 궁금합니다 :)

**Reply by anchangmin35** [reply]
넵 맞습니다! 
첫번째로 메서드 명을 예시로 들자면
public boolean check() 보다 public boolean isUserAuthenticated() 와 같이 어떤 역할을 하는지가 드러나는 것을 생각했습니다!

그리고 적절한 주석은 좀 포괄적으로 작성을 했었는데 
구체적으로는 모든 코드에 주석은 달 필요없지만 (오히려 가독성을 해칠 수 있다는 피드백을 들었던 것 같습니다!) 
핵심 로직이나 길어지는 코드 (남들이 봤을 때 어떤 코드인지 한 눈에 안들어오는 경우) 에는 간단하게라도 주석을 달자. 라는 의미였습니다!

**Comment by Daae-Kim** [comment]
좋은 의견, 답변 감사합니다 ㅎㅎ

그럼 모수 리팩토링에 앞서 좋은 코드의 기준은 
1. 코드의 중복을 줄인다.
2. 코드의 재사용성을 고려하여 작성한다.
3. 한개의 메소드는 한개의 역할을 한다.
4. 남이 보기에 쉬운 코드로 작성한다. (복잡한 로직의 경우 주석 달기)
5. 변수명, 메소드명은 동사 위주로 작명하고, 역할을 명확히 하는 명사를 사용한다.
6. 기능별로 코드를 분리하여 관심사를 분리한다.

이 6가지로 이야기가 된것 같습니다.
좋은 코드에 대한 이야기는 모수가 발전함에따라 적절히 변화할것이라고 생각합니다. 
그렇기에 이 디스커션은 열어두도록하고, 

모수 코드를 작성할때, 코드리뷰를 할때에 이 기준을 우선적으로 고려해주시면 감사하겠습니다.
잘부탁드립니다! 모수 화이팅~!!
